R/data-checks.R

```diff
@@ -1,25 +1,31 @@
 # Checkers ---------------------------------------------------------------------
-abort_bad_argument <- function(arg, must, not = NULL) {
-  msg <- glue::glue("`{arg}` must {must}")
+abort_bad_argument <- function(arg, must, not = NULL, extra = NULL,
+                               custom = NULL, call) {
+  msg <- glue::glue("{{.arg {arg}}} must {must}")
   if (!is.null(not)) {
     msg <- glue::glue("{msg}; not {not}")
   }
+  if (!is.null(extra)) {
+    msg <- c(msg, cli::format_message(extra))
```

Comment:
```suggestion
    msg <- c(msg, extra)
```

Better to keep `extra`  evaluation outside to avoid `.envir` problems. The problem could be solved by passing `.envir` for the evaluation of the message, but I didn't want to complicate things too much! I could send a follow-up to make this easier?

Reply:
Yes, this is perfect! This is why the `format_message()` inside `abort_bad_argument()` is redundant.